### DISCUSSION TITLE
Kendi class'im için sıralama algoritması

### DISCUSSION THREAD

**Discussion by BunyaminAkdeniz** [19/classroom#15] [Ruby] [answered]
Merhaba Hocam @roktas . Bir yerde nesnelerimi özelliklerine göre sıralamam gerekti. Fakat nesnelerin iki özelliği var.
Ornegin

```ruby

class Person 
  attr_reader :id, :second_id

  def initialize(id,second_id)
    @id = id 
    @second_id = second_id
  end

  def <=>(per)
    if per.id == id
      per.second_id <=> per
    else
      per.id <=> id
    end
  end 
end

b = Person.new(1,2)
a = Person.new(2,1)

list_of_person = [a, b]

list_of_person.sort 
```

Yapmak istediğim, bu iki person id'leri eşit ise second_id lerine göre sıralasın. Koşulu nasıl kuracağımı bilemedim

**Comment by roktas** [comment, marked as answer]
Yaklaşmışsın :-)  Eksik parça `Comparable` mixins

```ruby
class Person 
  include Comparable

  attr_reader :id, :second_id

  def initialize(id,second_id)
    @id = id 
    @second_id = second_id
  end

  def <=>(other)
    [id, second_id] <=> [other.id, other.second_id]
  end 
end

list_of_person = []

list_of_person << Person.new(1,2)
list_of_person << Person.new(2,1)
list_of_person << Person.new(1,1)

p list_of_person.sort 
```

Şu anki bilginle bu biraz ileri bir konu yalnız.  NYP dersinde gireceğim...

P.S. Mesajlarına kod gömerken dil belirtmeyi unutma, renklendirilmiş olur.  Şunu oku mutlaka: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by roktas** [reply]
`per` yerine geleneksel olarak `other` demeyi tercih ediyoruz.  İsmi hep böyle verirsen `<=>` metodları daha kolay okunur/anlaşılır.  Sıralamayı değiştirmek istersen sol ve sağ tarafları yer değiştir.

**Reply by BunyaminAkdeniz** [reply]
Teşekkür ederim Hocam. Söylediklerinize dikkat edeceğim
